sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts

```diff
@@ -274,115 +273,229 @@ export abstract class ECertFileHandler extends ESDCFileHandler {
-   * @param feedbackRecord E-Cert received record
+   * Get eCert feedback error code map which has
+   * error code as key and error id as value.
+   * The map is generated for the given offering intensity.
+   * @example {ERR01:1}.
+   * @param offeringIntensity offering intensity.
+   * @returns error code map.
    */
-  private async processErrorCodeRecords(
-    feedbackRecord: ECertFullTimeResponseRecord,
-  ): Promise<void> {
-    const disbursementSchedule =
-      await this.disbursementScheduleService.getDisbursementScheduleByDocumentNumber(
-        feedbackRecord.documentNumber,
+  private async getECertFeedbackErrorsMap(
+    offeringIntensity: OfferingIntensity,
+  ): Promise<Record<string, number>> {
+    const eCertFeedbackErrors =
+      await this.eCertFeedbackErrorService.getECertFeedbackErrorsByOfferingIntensity(
+        offeringIntensity,
       );
-    if (disbursementSchedule) {
-      await this.disbursementScheduleErrorsService.createECertErrorRecord(
-        disbursementSchedule,
+    const eCertFeedbackErrorCodeMap: Record<string, number> = {};
+    for (const eCertFeedbackError of eCertFeedbackErrors) {
+      eCertFeedbackErrorCodeMap[eCertFeedbackError.errorCode] =
+        eCertFeedbackError.id;
+    }
+    return eCertFeedbackErrorCodeMap;
+  }
+
+  /**
+   * Sanitize the e-Cert response records validating the error codes.
+   * @param eCertFeedbackResponseRecords e-Cert feedback response records to sanitize.
+   * @param eCertFeedbackErrorCodeMap e-Cert feedback error map
+   * to get error id by error code.
+   * @throws unknown error code error.
+   */
+  private sanitizeErrorCodes(
+    eCertFeedbackResponseRecords: ECertResponseRecord[],
+    eCertFeedbackErrorCodeMap: Record<string, number>,
+  ): void {
+    // Check for error codes sent that are not known to the system.
+    // In the case the system needs to be updated with latest error codes.
+    const unknownFeedbackErrorCodes: string[] =
+      eCertFeedbackResponseRecords.flatMap((eCertFeedbackResponseRecord) =>
         [
-          feedbackRecord.errorCode1,
-          feedbackRecord.errorCode2,
-          feedbackRecord.errorCode3,
-          feedbackRecord.errorCode4,
-          feedbackRecord.errorCode5,
-        ].filter((error) => error),
-        new Date(),
+          eCertFeedbackResponseRecord.errorCode1,
+          eCertFeedbackResponseRecord.errorCode2,
+          eCertFeedbackResponseRecord.errorCode3,
+          eCertFeedbackResponseRecord.errorCode4,
+          eCertFeedbackResponseRecord.errorCode5,
+        ].filter(
+          (errorCode) => errorCode && !eCertFeedbackErrorCodeMap[errorCode],
+        ),
       );
-    } else {
+    if (unknownFeedbackErrorCodes.length) {
       throw new Error(
-        `${feedbackRecord.documentNumber} document number not found in disbursement_schedule table.`,
+        `The following error codes are unknown to the system. ${Array.from(
+          new Set(unknownFeedbackErrorCodes),
```

Comment:
If the `unknownFeedbackErrorCodes` is already a `string[]` is it not possible to use the `join` directly?

Reply:
unknownFeedbackErrorCodes may have duplicate error codes and hence using set to get distinct error codes.